docs/developers/overview.mdx

```diff
@@ -23,8 +23,92 @@ modify states across different chains in a single step.
+
+Right now omnichain contracts support the following features:
+
+- Transfer of native gas asset (ETH, BNB, MATIC, etc.) between connected chains.
+- Transfer of ERC-20 tokens between connected chains.
+
+### Advantages of omnichain contracts
+
+Developers can extend common applications like Uniswap V2/V3, Curve, Aave, and
+Compound by making them compatible with ZRC-20. These applications have been
+audited and battle-tested on Ethereum and can easily be deployed and built on
+ZetaChain.The changes required to add ZRC-20 support are minimal, and users may
+interact with these applications in single-step transactions just as they would
+on Ethereum.
+
+ZRC-20 can easily support Bitcoin/Algorand/Cardano/XRP which do not have
+capability or efficiency to support general purpose smart contracts for
+applications like swapping, lending, etc.
+
+ZRC-20 may cost significantly less gas than message passing in many use cases
+because its interactions with foreign chains are confined to fungible coins
+(standard value transfer of ETH/BNB/MATIC costs 21k gas, moving ERC20 tokens
+costs around 60k gas; no logic/state on foreign chains means much less gas).
+
+With omnichain smart contracts, you can pair and trade native asset liquidity
+directly against each other. This minimizes steps to trade native assets in a
+single step, and it doesn't involve a bridge or wrapping step or sending complex
+messages. For example, you could trade Ethereum ETH directly for Polygon USDC
+through a unified pool in a single transaction.
+
+With ZRC-20/zEVM, exception/revert handling is much simpler as foreign
+interaction is confined to either a standard ERC-20/contract interaction
+(success) or no contract interaction (failure) immediately.
+
+## Cross-chain message passing
+
+With cross-chain message (CCM) passing you deploy **a CCM-enabled contract on
+each connected chain**.
+
+On a connected chain a user can call a CCM-enabled contract. The contract calls
+the Connector API and sends a message. ZetaChain acts as relayer and transmits
+the message to the destination chain. On the destination chain a CCM-enabled
+contract uses the Connector API to receive the message and handles it.
+
+The state is stored on a set of CCM-enabled contracts on different chains.
+
+CCM makes sense for applications that only need unidirectional and asynchronous
+logic/effects and that don't need or benefit from a unified state.
+
+### Advantages of cross-chain message passing
+
+CCM supports any type of data transfer and it's up to the contracts to decide
+how to handle the data.
+
+CCM is a reliable general purpose solution for augmenting existing applications
+with cross-chain functionality.
+
+With messages, you can leverage existing liquidity like Uniswap pools on
+existing chains, and trade via burning/minting ZETA through ZetaChain. This
+approach may be more complex (more transactions involved, more gas), but doesn't
+rely on liquidity within ZetaChain's ecosystem.
```

Comment:
add-on sentence:

`By leveraging the ZETA burn/mint functionality built into ZetaChain, one can build value-transfer applications that don't bridge/wrap assets, minimizing risk for users.`

Reply:
857346b